### PR TITLE
Add extra configuration for .codacy.json file

### PR DESCRIPTION
**Pull request by mrfyda**

### CODE REVIEW

File: src/main/scala/codacy/docker/api/package.scala [hunk 1]
```diff
@@ -11,4 +11,9 @@ package object api extends JsonApi{
     def Value(raw:String):Parameter.Value = Value(Try(Json.parse(raw)).getOrElse(JsString(raw)))
   }
 
-}
\ No newline at end of file
+  implicit class ConfigurationExtensions(config:Configuration.type){
+    def Value(jsValue:JsValue):Configuration.Value = ConfigurationValue(jsValue)
```
Comment (rtfpessoa) [inline src/main/scala/codacy/docker/api/package.scala]: y u n format?

Comment (mrfyda) [inline src/main/scala/codacy/docker/api/package.scala]: Intellij 2017.1 applies contextual format depending on the file previous formatting.

Comment (rtfpessoa) [inline src/main/scala/codacy/docker/api/package.scala]: The force him to format correctly

File: src/main/scala/codacy/dockerApi/DockerEngine.scala [hunk 2]
```diff
@@ -61,10 +61,12 @@ abstract class DockerEngine(Tool: codacy.docker.api.Tool) extends Delayed {
         Tool.apply(
           source = Source.Directory(sourcePath.toString()),
           configuration = patternsOpt,
-          files = filesOpt
+          files = filesOpt,
+          options = configOpt.flatMap(_.options).getOrElse(Map.empty)
         )(spec)
       } catch {
         case t: Throwable =>
+          t.printStackTrace(Console.err)
```
Comment (rtfpessoa) [inline src/main/scala/codacy/dockerApi/DockerEngine.scala]: Why would we want this?

Comment (mrfyda) [inline src/main/scala/codacy/dockerApi/DockerEngine.scala]: To debug when something goes wrong.

Comment (rtfpessoa) [inline src/main/scala/codacy/dockerApi/DockerEngine.scala]: But the value is being returned. The person receiving it should decide. Here does not seem a good place to print values without option to remove verbosity.

**Pull request merged**